src/frontends/pytorch/src/op/avg_poolnd.cpp

```diff
@@ -8,19 +8,39 @@
 #include "openvino/op/concat.hpp"
 #include "openvino/op/constant.hpp"
 #include "openvino/op/pad.hpp"
-#include "openvino/op/subtract.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/slice.hpp"
+#include "openvino/op/unsqueeze.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/add.hpp"
 #include "utils.hpp"
 
+
 namespace ov {
 namespace frontend {
 namespace pytorch {
 namespace op {
 
 using namespace ov::op;
-
-OutputVector translate_avg_poolnd(const NodeContext& context) {
+OutputVector translate_avg_pool_base(const NodeContext& context, int dims) {
     num_inputs_check(context, 2, 7);
     auto input = context.get_input(0);
+    auto input_shape = context.mark_node(std::make_shared<v3::ShapeOf>(input));
+
+    auto const_0 = v0::Constant::create(element::i64, Shape{1}, {0});
+    auto const_1 = v0::Constant::create(element::i64, Shape{1}, {1});
+
+    // Unsqueeze axis to add batch dimension
+    input = context.mark_node(std::make_shared<v0::Unsqueeze>(input, const_0));
```

Comment:
```suggestion
    if (input.get_partial_shape().rank().is_static()) {
        if (input.get_partial_shape().rank().get_length() == dims + 1) {
            input = context.mark_node(std::make_shared<v0::Unsqueeze>(input, const_0));
        }
    } else {
        // Process dynamic case here
    }
```
This is only needed when you have static rank, in case of dynamic please update else branch to use Reshape in there.
Also do the same for output